workflow/controller/operator.go

```diff
@@ -2186,3 +2205,64 @@ func (woc *wfOperationCtx) runOnExitNode(parentName, templateRef, boundaryID str
 	}
 	return false, nil, nil
 }
+
+func (woc *wfOperationCtx) createPDBResource() error {
+
+	if woc.wf.Spec.PodDisruptionBudget == nil {
+		return nil
+	}
+
+	pdb, err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Get(woc.wf.Name, metav1.GetOptions{})
+	if err != nil && !apierr.IsNotFound(err) {
+		return err
+	}
+	if pdb != nil && pdb.Name != "" {
+		return nil
+	}
+
+	if woc.wf.Spec.PodDisruptionBudget.Selector == nil {
+		woc.wf.Spec.PodDisruptionBudget.Selector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{common.LabelKeyWorkflow: woc.wf.Name},
+		}
+	}
+
+	newPDB := policyv1beta.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   woc.wf.Name,
+			Labels: map[string]string{common.LabelKeyWorkflow: woc.wf.Name},
+		},
+		Spec: *woc.wf.Spec.PodDisruptionBudget,
+	}
+	newPDB.SetOwnerReferences(append(newPDB.GetOwnerReferences(), *metav1.NewControllerRef(woc.wf, wfv1.SchemeGroupVersion.WithKind(workflow.WorkflowKind))))
+	created, err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Create(&newPDB)
+	if err != nil {
+		return err
+	}
+	woc.log.Infof("PDB resource %s created for workflow %s", created.Name, woc.wf.Name)
+	woc.updated = true
+	return nil
+}
+
+func (woc *wfOperationCtx) deletePDBResource() error {
+	if woc.wf.Spec.PodDisruptionBudget == nil {
+		return nil
+	}
+	var err error
+	_ = wait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
```

Comment:
you should not ignore error here, line 2256 would be ignored

Reply:
Error is not ignored.  error is assing outside variable and handled below. `wait.ExponentialBackoff` will have a little bit of different implementation to return an error.
`var err error`